de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/dataaccess/RepositoryUtility.java

```diff
@@ -0,0 +1,191 @@
+import de.dlr.sc.virsat.model.dvlm.tree.ITreeTraverserMatcher;
+import de.dlr.sc.virsat.model.dvlm.tree.TreeTraverser;
+
+public class RepositoryUtility {
+
+	private RepositoryUtility() { }
+
+	/**
+	 * Traverses a Repository and all Seis
+	 */
+	static class UuidTraverser implements ITreeTraverserMatcher<IUuid> {
+
+		@Override
+		public boolean isMatching(IUuid treeNode) {
+			return false;
+		}
+
+		@Override
+		public void processMatch(IUuid treeNode, IUuid matchingParent) {
+			// Will be handled in isMatching
+		}
+
+		@Override
+		public Collection<? extends IUuid> getChildren(IUuid treeNode) {
+			if (treeNode instanceof Repository) {
+				return ((Repository) treeNode).getRootEntities();
+			} else if (treeNode instanceof StructuralElementInstance) {
+				return ((StructuralElementInstance) treeNode).getChildren();
+			}
+			return null;
+		}
+		
+	}
+	
+	/**
+	 * Finds a sei instance by it's uuid
+	 * @param uuid the seis uuid
+	 * @return the StructuralElementInstance or null
+	 * @throws CoreException
+	 */
+	public static StructuralElementInstance findSei(String uuid, Repository repository) throws CoreException {
+		
+		List<StructuralElementInstance> match = new ArrayList<StructuralElementInstance>(); 
+		
+		TreeTraverser<IUuid> traverser = new TreeTraverser<IUuid>();
+		traverser.traverse(repository, new UuidTraverser() {
+			@Override
+			public boolean isMatching(IUuid treeNode) {
+				boolean isMatching = treeNode.getUuid().toString().equals(uuid);
+				
+				if (isMatching) {
+					match.add((StructuralElementInstance) treeNode);
+				}
+				
+				return isMatching;
+			}
+		});
+		
+		if (match.isEmpty()) {
+			return null;
+		} else {
+			return match.get(0);
+		}
+	}
+	
+	/**
+	 * Iterates over all seis in the repository and matches the uuid on eobject level
+	 * @param uuid
+	 * @param repository
+	 * @return
+	 */
+	public static IUuid findObjectById(String uuid, Repository repository) {
+		List<IUuid> match = new ArrayList<IUuid>(); 
+		
+		TreeTraverser<IUuid> traverser = new TreeTraverser<IUuid>();
+		traverser.traverse(repository, new UuidTraverser() {
+			@Override
+			public boolean isMatching(IUuid treeNode) {
+				boolean isMatching = false;
+				
+				if (treeNode instanceof StructuralElementInstance) {
+					EObject objByUuid = treeNode.eResource().getEObject(uuid);
+					isMatching = objByUuid != null;
+					if (isMatching) {
+						match.add((IUuid) objByUuid);
+					}
+				}
+				
+				return isMatching;
+			}
+		});
+		
+		if (match.isEmpty()) {
+			return null;
+		} else {
+			return match.get(0);
+		}
+	}
+	
+	/**
+	 * Finds a ca instance by it's uuid
+	 * @param uuid the cas uuid
+	 * @return the CategoryAssignment or null
+	 * @throws CoreException
+	 */
+	public static CategoryAssignment findCa(String uuid, Repository repository) {
+		IUuid obj = findObjectById(uuid, repository);
+		if (obj instanceof CategoryAssignment) {
+			return (CategoryAssignment) obj;
+		}
+		
+		return null;
```

Comment:
There generaly should also be test cases that do not find the requested object.

Reply:
I added the tests